exoctk/utils.py

```diff
@@ -36,6 +34,45 @@
 VERSION = '0.2'
 
 
+def color_gen(colormap='viridis', key=None, n=10):
+    """Color generator for Bokeh plots
+
+    Parameters
+    ----------
+    colormap: str, sequence
+        The name of the color map
+
+    Returns
+    -------
+    generator
+        A generator for the color palette
+    """
+    if colormap in dir(bpal):
+        palette = getattr(bpal, colormap)
+
+        if isinstance(palette, dict):
+            if key is None:
+                key = list(palette.keys())[0]
+            palette = palette[key]
+
+        elif callable(palette):
+            palette = palette(n)
+
+        else:
+            raise TypeError("pallette must be a bokeh palette name or a sequence of color hex values.")
+
+    elif isinstance(colormap, (list, tuple)):
+        palette = colormap
+
+    else:
+        raise TypeError("pallette must be a bokeh palette name or a sequence of color hex values.")
+
+    yield from itertools.cycle(palette)
+
+
+COLORS = color_gen('Category10')
+
+
```

Comment:
@hover2pi I do like this `viridis` hack -- but I would suspect there must be an easier way to import this color scheme?

Reply:
There isn't! It's a serious flaw of `bokeh`. This function I added is a way to turn a `bokeh` color palette into a generator using `itertools` so that we can just call `next` on an arbitrary number of plot elements without getting an `IndexError`.